src/sink/amqp.rs

```diff
@@ -0,0 +1,268 @@
+// Copyright 2020-2021, The Tremor Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![cfg(not(tarpaulin_include))]
+
+//! # AMQP Offramp
+//!
+//! The `amqp` offramp allows producing events to an amqp broker.
+
+use crate::sink::prelude::*;
+use async_channel::{bounded, Receiver};
+use serde::{Deserialize};
+use crate::url::TremorUrl;
+use lapin::{
+    options::*, Connection, ConnectionProperties, Channel, BasicProperties, publisher_confirm::Confirmation
+};
+use halfbrown::HashMap;
+use std::{
+    fmt,
+    time::{Instant},
+};
+
+#[derive(Deserialize, Debug, Clone)]
+pub struct Config {
+    pub amqp_addr: String,
+    routing_key: String,
+    exchange: String,  // "", by default
```

Comment:
Adding
```rust
    #[serde(default = "Default::default")]`
```
to this would make it optional and default to `""` when nothing is provided

Reply:
that's a great idea, I checked that both routing_key and exchange can be blank in amqp, which is the default and in fanout mode even is of no use (so blank string it is).